src/crimson/os/seastore/transaction.h

```diff
@@ -141,8 +141,25 @@ class Transaction {
     return retired_set;
   }
 
+  enum class src_t : uint8_t {
+    SEASTORE_MUTATE = 0,
+    SEASTORE_READ,
+    CLEANER,
+    // including tests and tools that bypass seastore
+    TEST,
```

Comment:
I also don't really like TEST.  The same scenario won't have both TEST and MUTATE (for instance) operations, so distinguishing between them isn't useful.  Distinguishing between MUTATE, READ, and CLEANER within a test, however, is potentially useful.

Reply:
see https://github.com/ceph/ceph/pull/42202#discussion_r666634114

Reply:
I disagree.  I don't see any reason why opertations originating from tests can't be tagged as READ/WRITE/CLEANER.